test/functional/repository/reload/repository-reload.ts

```diff
@@ -0,0 +1,54 @@
+import "reflect-metadata";
+import {closeTestingConnections, createTestingConnections, reloadTestingDatabases} from "../../../utils/test-utils";
+import {Connection} from "../../../../src/connection/Connection";
+import {Post} from "./entity/Post";
+import {Category} from "./entity/Category";
+
+describe("repository > reload method", () => {
+
+    let connections: Connection[];
+    before(async () => connections = await createTestingConnections({
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+    }));
+    beforeEach(() => reloadTestingDatabases(connections));
+    after(() => closeTestingConnections(connections));
+
+    it("should reload given entity", () => Promise.all(connections.map(async connection => {
+        Post.useConnection(connection);
+        Category.useConnection(connection); 
+
+        const category = Category.create();
+        category.name = "Persistence";
+        await category.save();
+
+        const post = Post.create();
+        post.title = "About ActiveRecord";
+        post.categories = [category];
+        await post.save();
+
+        await post.reload();
+    
+        const assertCategory = Object.assign({}, post.categories[0]);
+        post!.should.be.instanceOf(Post);
+        post!.id.should.be.eql(1);
+        post!.title.should.be.eql("About ActiveRecord");
+        post!.text.should.be.eql("This is default text.");
+        assertCategory.should.be.eql({
+            id: 1,
+            name: "Persistence"
+        });
+
+        category.name = "Persistence and Entity";
+        await category.save();
+        
+        await post.reload();
```

Comment:
I don't like activeRecord-like test when test is about `repository > reload method`, but I am not the boss here :)

Reply:
Yes, I wasn't sure about this either. I think I'll make a repository test instead